confmap/confmap.go

```diff
@@ -57,19 +57,55 @@ func (l *Conf) AllKeys() []string {
 	return l.k.Keys()
 }
 
-// Unmarshal unmarshalls the config into a struct.
+type UnmarshalOption interface {
+	apply(*unmarshalOption)
+}
+
+type unmarshalOption struct {
+	// If ErrorUnused is true, then it is an error for there to exist
+	// keys in the original Conf that were unused in the decoding process
+	// (extra keys).
+	errorUnused bool
+}
+
+// WithErrorUnused  then it is an error for there to exist
+// keys in the original Conf that were unused in the decoding process
+// (extra keys).
+func WithErrorUnused() UnmarshalOption {
```

Comment:
Question if this should accept a bool or we allow only to set. This feels ok to me as is.